src/main/java/de/tum/in/www1/artemis/service/competency/CourseCompetencyService.java

```diff
@@ -63,6 +114,291 @@ public CourseCompetencyService(CompetencyProgressRepository competencyProgressRe
+        if (course.getLearningPathsEnabled()) {
+            var importedCompetencies = idToImportedCompetency.values().stream().map(CompetencyWithTailRelationDTO::competency).toList();
+            learningPathService.linkCompetenciesToLearningPathsOfCourse(importedCompetencies, course.getId());
+        }
+
+        var originalCompetencyIds = idToImportedCompetency.keySet();
+        var relations = competencyRelationRepository.findAllByHeadCompetencyIdInAndTailCompetencyIdIn(originalCompetencyIds, originalCompetencyIds);
+
+        if (!relations.isEmpty()) {
+            for (var relation : relations) {
+                var tailCompetencyDTO = idToImportedCompetency.get(relation.getTailCompetency().getId());
+                var headCompetencyDTO = idToImportedCompetency.get(relation.getHeadCompetency().getId());
+
+                CompetencyRelation relationToImport = new CompetencyRelation();
+                relationToImport.setType(relation.getType());
+                relationToImport.setTailCompetency(tailCompetencyDTO.competency());
+                relationToImport.setHeadCompetency(headCompetencyDTO.competency());
+
+                relationToImport = competencyRelationRepository.save(relationToImport);
+                tailCompetencyDTO.tailRelations().add(CompetencyRelationDTO.of(relationToImport));
+            }
+        }
+        return new HashSet<>(idToImportedCompetency.values());
+    }
+
+    /**
+     * Imports the standardized competencies with the given ids as course competencies into a course
+     *
+     * @param competencyIdsToImport    the ids of the standardized competencies to import
+     * @param course                   the course to import into
+     * @param courseCompetencySupplier the supplier for creating new course competencies
+     * @return the list of imported competencies
+     */
+    public List<CourseCompetency> importStandardizedCompetencies(List<Long> competencyIdsToImport, Course course, Supplier<CourseCompetency> courseCompetencySupplier) {
+        List<StandardizedCompetency> standardizedCompetencies = standardizedCompetencyRepository.findAllById(competencyIdsToImport);
+
+        if (standardizedCompetencies.size() != competencyIdsToImport.size()) {
+            throw new EntityNotFoundException("Could not find all standardized competencies to import in the database!");
+        }
+
+        List<CourseCompetency> competenciesToCreate = new ArrayList<>();
+
+        for (var standardizedCompetency : standardizedCompetencies) {
+            var competency = courseCompetencySupplier.get();
+            competency.setTitle(standardizedCompetency.getTitle());
+            competency.setDescription(standardizedCompetency.getDescription());
+            competency.setTaxonomy(standardizedCompetency.getTaxonomy());
+            competency.setMasteryThreshold(Competency.DEFAULT_MASTERY_THRESHOLD);
```

Comment:
Is it sensible to always set the mastery threshold to the default value instead of the value that is defined in the imported competency?